src/sirinternal.c

```diff
@@ -59,12 +59,16 @@ static sir_plugincache _sir_pc = {0};
 static sir_mutex cfg_mutex  = SIR_MUTEX_INIT;
 static sir_mutex fc_mutex   = SIR_MUTEX_INIT;
 static sir_mutex pc_mutex   = SIR_MUTEX_INIT;
+# if !defined(SIR_NO_TEXT_STYLING)
 static sir_mutex ts_mutex   = SIR_MUTEX_INIT;
+# endif
 #else
 static sir_mutex cfg_mutex  = {0};
```

Comment:
tbh, i would rather set all of these to `= {0}` here, OR if we can get away with no warnings, just don't initialize them at all and delete `SIR_MUTEX_INIT`...

Since on everything but Windows `SIR_MUTEX_INIT` is `PTHREAD_MUTEX_INITIALIZER` which expands to `#define PTHREAD_MUTEX_INITIALIZER {_PTHREAD_MUTEX_SIG_init, {0}}` and `_PTHREAD_MUTEX_SIG_init` is a magic number, and it is still working with importc and pthreads, it may not even be necessary to use `PTHREAD_MUTEX_INITIALIZER` at all, so these can always just be `= {0}` and `SIR_MUTEX_INIT` can go adios

Reply:
Worth a shot. We will know rather quickly if `pthread_create` fails.

Reply:
It gave me warnings ... I'll mess with this in another PR.

Reply:
ya rite, we will forget.

Reply:
I'd have to check, but POSIX says we should do it.

"PTHREAD_MUTEX_INITIALIZER can be used to initialize any mutexes that are statically allocated. The effect shall be equivalent to dynamic initialization by a call to thread_mutex_init() with parameter attr specified as NULL, except that no error check is performed."